### PR TITLE
add partition_key_version attribute to resource azurerm_cosmosdb_sql_container

### PR DESCRIPTION
**Pull request by ankushjain3005**
Azure Cosmos DB now supports large partition keys with values up to 2 KB. This pull request captures adding an attribute partition_key_version which can be used to set this configuration. This [URL](https://docs.microsoft.com/en-us/azure/cosmos-db/large-partition-keys?tabs=dotnetv3) has more details about this.
This closes [#8653](https://github.com/terraform-providers/terraform-provider-azurerm/issues/8653).

### CODE REVIEW

**Comment by ankushjain3005** [comment]
Test Result:

`
ankujai@Ankujai-lp MINGW64 ~/go/src/github.com/terraform-providers/terraform-provider-azurerm (master)
$ TF_ACC=1 go test -v ./azurerm/internal/services/cosmos/ -run TestAccAzureRMCosmosDbSqlContainer -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"                === RUN   TestAccAzureRMCosmosDbSqlContainer_basic
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_basic
=== RUN   TestAccAzureRMCosmosDbSqlContainer_basic_serverless
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_basic_serverless
=== RUN   TestAccAzureRMCosmosDbSqlContainer_complete
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_complete
=== RUN   TestAccAzureRMCosmosDbSqlContainer_update
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_update
=== RUN   TestAccAzureRMCosmosDbSqlContainer_autoscale
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_autoscale
=== RUN   TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== RUN   TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== CONT  TestAccAzureRMCosmosDbSqlContainer_basic
=== CONT  TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== CONT  TestAccAzureRMCosmosDbSqlContainer_autoscale
=== CONT  TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== CONT  TestAccAzureRMCosmosDbSqlContainer_complete
=== CONT  TestAccAzureRMCosmosDbSqlContainer_update
=== CONT  TestAccAzureRMCosmosDbSqlContainer_basic_serverless
--- PASS: TestAccAzureRMCosmosDbSqlContainer_basic_serverless (1077.73s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_basic (1090.11s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_partition_key_version (1159.99s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_complete (1196.99s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_update (1303.25s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_indexing_policy (1424.73s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_autoscale (1562.54s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos      1572.680s

`

File: azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go [hunk 1]
```diff
@@ -158,9 +166,17 @@ func resourceArmCosmosDbSQLContainerCreate(d *schema.ResourceData, meta interfac
...
+				Kind:    documentdb.PartitionKindHash,
+				Version: utils.Int32(int32(partitionKeyVersion.(int))),
+			}
+		} else {
+			db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
+				Paths: &[]string{partitionkeypaths},
+				Kind:  documentdb.PartitionKindHash,
+			}
 		}
 	}
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go:181]: We could make this something like
```suggestion
	if partitionkeypaths != "" {
		db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
			Paths: &[]string{partitionkeypaths},
			Kind:  documentdb.PartitionKindHash,
		}
		
		if v, ok := d.GetOk("partition_key_version"); ok {
			db.SQLContainerCreateUpdateProperties.Resource.PartitionKey.Version = utils.Int32(int32(partitionKeyVersion.(int))),
		}
	}
```

File: azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go [hunk 2]
```diff
@@ -241,9 +258,17 @@ func resourceArmCosmosDbSQLContainerUpdate(d *schema.ResourceData, meta interfac
...
 
 	if partitionkeypaths != "" {
-		db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go:244]: Same as above

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ankushjain3005 - overall this looks good, once the comments i've left inline are addressed (and testes pass) this should be good to merge!

File: azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go (hunk 1 above)
Comment (ankushjain3005) [reply azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go]: resolved

File: azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go (hunk 2 above)
Comment (ankushjain3005) [reply azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go]: resolved

**Comment by ankushjain3005** [comment]
resolved the comments and reran the test. here are the results

$ TF_ACC=1 go test -v ./azurerm/internal/services/cosmos/ -run TestAccAzureRMCosmosDbSqlContainer -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc" === RUN
go: downloading github.com/tombuildsstuff/giovanni v0.15.0
=== RUN   TestAccAzureRMCosmosDbSqlContainer_basic
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_basic
=== RUN   TestAccAzureRMCosmosDbSqlContainer_basic_serverless
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_basic_serverless
=== RUN   TestAccAzureRMCosmosDbSqlContainer_complete
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_complete
=== RUN   TestAccAzureRMCosmosDbSqlContainer_update
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_update
=== RUN   TestAccAzureRMCosmosDbSqlContainer_autoscale
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_autoscale
=== RUN   TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== RUN   TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== PAUSE TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== CONT  TestAccAzureRMCosmosDbSqlContainer_basic
=== CONT  TestAccAzureRMCosmosDbSqlContainer_autoscale
=== CONT  TestAccAzureRMCosmosDbSqlContainer_update
=== CONT  TestAccAzureRMCosmosDbSqlContainer_indexing_policy
=== CONT  TestAccAzureRMCosmosDbSqlContainer_complete
=== CONT  TestAccAzureRMCosmosDbSqlContainer_partition_key_version
=== CONT  TestAccAzureRMCosmosDbSqlContainer_basic_serverless
--- PASS: TestAccAzureRMCosmosDbSqlContainer_partition_key_version (1369.94s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_complete (1369.94s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_basic_serverless (1369.95s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_basic (1369.98s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_indexing_policy (1398.07s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_update (1401.23s)
--- PASS: TestAccAzureRMCosmosDbSqlContainer_autoscale (1885.70s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos      1898.769s

**Review by katbyte — APPROVED** [review]
Thanks @ankushjain3005 - LGTM 👍

**Pull request merged**